Gemfile

```diff
@@ -136,22 +152,31 @@ gem 'tzinfo', '>= 1.2.0'	# required by rails; 1.2.0 has support for *BSD and Sol
 # Windows does not have zoneinfo files, so bundle the tzinfo-data gem.
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw]
 
-# Introduces a scope for Heroku specific gems.
-def on_heroku
-  if ENV['ON_HEROKU'] ||
-     ENV['HEROKU_POSTGRESQL_ROSE_URL'] ||
-     ENV['HEROKU_POSTGRESQL_GOLD_URL'] ||
-     File.read(File.join(File.dirname(__FILE__), 'Procfile')) =~ /intended for Heroku/
+
+# Introduces a scope for gem loading based on a condition
+def if_true(condition)
+  if condition
     yield
   else
-    # When not on Heroku, we still want our Gemfile.lock to include
-    # Heroku specific gems, so we scope them to an unsupported
-    # platform.
+    # When not including the gems, we still want our Gemfile.lock
+    # to include them, so we scope them to an unsupported platform.
     platform :ruby_18, &proc
   end
 end
 
-on_heroku do
-  gem 'pg'
+on_heroku = ENV['ON_HEROKU'] ||
+            ENV['HEROKU_POSTGRESQL_ROSE_URL'] ||
+            ENV['HEROKU_POSTGRESQL_GOLD_URL'] ||
+            File.read(File.join(File.dirname(__FILE__), 'Procfile')) =~ /intended for Heroku/
+
```

Comment:
Maybe you could say `ENV['DATABASE_ADAPTER'] ||= 'postgresql' if on_heroku` and change the condition `on_heroku || ENV['DATABASE_ADAPTER'].strip == 'postgresql'` below to just `ENV['DATABASE_ADAPTER'].strip == 'postgresql'`, so user could choose mysql2 even on Heroku by predefining DATABASE_ADAPTER.